src/main/java/org/spongepowered/common/mixin/core/entity/item/MixinEntityItem.java

```diff
@@ -26,156 +26,47 @@
-            shift = At.Shift.AFTER))
-    private void onOnUpdateAge(CallbackInfo ci) {
-        if (this.delayBeforeCanPickup == MAGIC_INFINITE_DESPAWN_TIME && !this.infiniteDespawnDelay && this.pluginDespawnSet) {
-            this.delayBeforeCanPickup--;
-        }
-    }
-
+    @Override
     public int getPickupDelay() {
-        if (this.delayBeforeCanPickup == MAGIC_INFINITE_PICKUP_DELAY) {
-            // There are two cases when -1 should be returned:
-
-            // The plugin has called set an infinite pickup delay
-            // The plugin has not set a pickup delay (neither setPickupDelay nor setInfinitePickupDelay
-            // has been called) - a Forge mod or something else has set the pickup delay, and they presumably
-            // know about the magic value.
-            if ((this.pluginPickupSet && this.infinitePickupDelay) || !this.pluginPickupSet) {
-                return MAGIC_INFINITE;
-            }
-        }
         return this.delayBeforeCanPickup;
     }
 
-    public void setPickupDelay(int delay) {
-        this.delayBeforeCanPickup = delay;
-        this.pluginPickupSet = true;
-        this.infinitePickupDelay = false;
-    }
-
-    public void setInfinitePickupDelay() {
-        this.delayBeforeCanPickup = MAGIC_INFINITE_PICKUP_DELAY;
-        this.pluginPickupSet = true;
-        this.infinitePickupDelay = true;
-    }
-
-    public int getDespawnTime() {
-        if (this.age == MAGIC_INFINITE_DESPAWN_TIME) {
-            if ((this.pluginDespawnSet && this.infiniteDespawnDelay) || !this.pluginDespawnSet) {
-                return MAGIC_INFINITE;
-            }
-        }
-        return this.lifespan - this.age;
-    }
-
-    public void setDespawnTime(int time) {
-        this.lifespan = this.age + time;
-        this.pluginDespawnSet = true;
-        this.infiniteDespawnDelay = false;
-    }
-
-    public void setInfiniteDespawnTime() {
-        this.age = MAGIC_INFINITE_DESPAWN_TIME;
-        this.pluginDespawnSet = true;
-        this.infiniteDespawnDelay = true;
-    }
-
     @Override
-    public void readFromNbt(NBTTagCompound compound) {
-        super.readFromNbt(compound);
-        // If the key exists, the value has been set by a plugin
-        if (compound.hasKey("infinitePickupDelay")) {
-            this.pluginPickupSet = true;
-            if (compound.getBoolean("infinitePickupDelay")) {
-                this.setInfinitePickupDelay();
-            } else {
-                this.infinitePickupDelay = false;
-            }
-        }
-        if (compound.hasKey("infiniteDespawnDelay")) {
-            this.pluginDespawnSet = true;
-            if (compound.getBoolean("infiniteDespawnDelay")) {
-                this.setInfiniteDespawnTime();
-            } else {
-                this.infiniteDespawnDelay = false;
-            }
-        }
+    public int getDespawnDelay() {
+        return this.age;
     }
 
     @Override
-    public void writeToNbt(NBTTagCompound compound) {
```

Comment:
The original intention was to avoid magic values
https://github.com/SpongePowered/SpongeAPI/pull/338#discussion_r22431290